### PR TITLE
Remove RPC_SOC and cl_geterr, plus rpc_raw dummy

### PR DESCRIPTION
**Pull request by was4**
Remove obsolete RPC_SOC definitions
rpc_raw dummy
Remove cl_geterr

### CODE REVIEW

File: src/clnt_generic.c [hunk 1]
```diff
@@ -517,11 +503,12 @@ clnt_req_setup(struct clnt_req *cc, struct timespec timeout)
...
 			"%s: %p fd %d insert failed xid %" PRIu32,
 			__func__, &rec->xprt, rec->xprt.xp_fd, cc->cc_xid);
-		return (false);
+		cc->cc_error.re_status = RPC_TLIERROR;
+		return (RPC_TLIERROR);
 	}
 
 	CLNT_REF(clnt, CLNT_REF_FLAG_NONE);
-	return (true);
+	return (RPC_SUCCESS);
```
Comment (dang) [inline src/clnt_generic.c:529]: This path doesn't set cc_error.  RPC_SUCCESS is zero, so it should be okay, but in the function above, you explicitly set cc_error in the success case.  We should probably stick to one form or the other.

File: src/pmap_getport.c [hunk 2]
```diff
@@ -71,22 +71,29 @@ u_short pmap_getport(struct sockaddr_in *address, u_long program,
...
-				clnt_geterr(client, &rpc_createerr.cf_error);
-			} else if (port == 0) {
-				rpc_createerr.cf_stat = RPC_PROGNOTREGISTERED;
-			}
+		stat = clnt_req_setup(cc, to);
+		if (stat == RPC_SUCCESS) {
+			stat = CLNT_CALL_WAIT(cc);
+		}
+		if (stat != RPC_SUCCESS) {
+			char *t = rpc_sperror(&cc->cc_error, __func__);
```
Comment (dang) [inline src/pmap_getport.c]: This section seems to pop up a lot.  Maybe an inline function or macro?  I suspect a free will be missed at some point.

**Review by dang — COMMENTED** [review]
This looks like a very nice cleanup.  I'll appove once it's ready.

**Comment by was4** [comment]
On 12/13/17 2:03 PM, Daniel Gryniewicz wrote:
> In src/clnt_generic.c <https://github.com/nfs-ganesha/ntirpc/pull/96#discussion_r156743026>:
> 
>>  	}
>   
>   	CLNT_REF(clnt, CLNT_REF_FLAG_NONE);
> -	return (true);
> +	return (RPC_SUCCESS);
> 
> This path doesn't set cc_error. RPC_SUCCESS is zero, so it should be okay, but in the function above, you explicitly set cc_error in the success case. We should probably stick to one form or the other.
> 
Yes it does, I just didn't move the old line. This is line 511,
you'll find the cc_error default above at 475-476. Because this
isn't guaranteed to be calloc (almost always malloc), have to
set everything as needed.

**Comment by was4** [comment]
On 12/13/17 2:03 PM, Daniel Gryniewicz wrote:
> In src/pmap_getport.c <https://github.com/nfs-ganesha/ntirpc/pull/96#discussion_r156744627>:
> This section seems to pop up a lot. Maybe an inline function or macro? I suspect a free will be missed at some point.
> 
In the original patch, I tried to make the patch easy to review by
keeping the same willy-nilly order of the original coding.  Now
I'm going through and narrowing down to 2 consistent patterns.

Yes, I've found 3 or 4 paths so far where a free was missed, and
fixed them.  Consistent patterns helped.

Also having patterns allowed me to show that all this code is
duplicated in rpcb_clnt, so we can throw this file away.

**Comment by was4** [comment]
On 12/13/17 2:03 PM, Daniel Gryniewicz wrote:
> This looks like a very nice cleanup. I'll appove once it's ready.
> 
I spent a lot of extra time doing cleanup.  But this is about
correctness.  The cl_geterr and clnt_sperror were fetching per
client re_status, but the actual returns are now usually in the
clnt_req re_status instead.  My logs were showing "success" on
failure lines.

**Review by dang — APPROVED** [review]

**Pull request merged**